---
title: "Analysis of ctDNA Data from GIST  Phase I Study"
author: "Stat4ward, `r format(Sys.time(), '%d %B, %Y')`"
output:
  word_document:
     reference_docx: reference_doc.docx
toc: TRUE
---
\pagebreak

# GOAL

# DATA

```{r setup}
knitr::opts_chunk$set(cache=FALSE,echo = FALSE, warning=FALSE, message=FALSE, fig.width = 6.5, fig.height = 6.5)

rm(list = ls())
library(haven)
library(tidyverse)
library(readr)
library(ggplot2)
library(knitr)
library(readxl)
library(purrr)
library(tableone)

# data input

adrs<- read_sas("C:\\Users\\ShaYang\\OneDrive - Stat4ward\\Documents\\Deciphera\\Phase I\\GIST\\adrs.sas7bdat")
#glimpse(adrs)

adsl<- read_sas("C:\\Users\\ShaYang\\OneDrive - Stat4ward\\Documents\\Deciphera\\Phase I\\GIST\\adsl.sas7bdat")
#glimpse(adsl)

adtte<-read_sas("C:\\Users\\ShaYang\\OneDrive - Stat4ward\\Documents\\Deciphera\\Phase I\\GIST\\adtte.sas7bdat")
#glimpse(adtte)

ctDNA.all<-read.csv("C:\\Users\\ShaYang\\OneDrive - Stat4ward\\Documents\\Deciphera\\Phase I\\GIST\\20210319_Deciphera_SOW1_1064Samples_G360Report_CumulativeLATEST.csv")

ctDNA<-ctDNA.all
# %>%filter(Cancertype=="GIST (Gastrointestinal stromal tumor)"|Cancertype=="GIST (Gastrointestinal Stromal Tumor)")

successful.sample.list<-read_excel("C:\\Users\\ShaYang\\OneDrive - Stat4ward\\Documents\\Deciphera\\Phase I\\GIST\\Copy of 20210209_Deciphera_SOW1_1064Samples_G360Report_Successful_Failed.xlsx", sheet="Successful ")

successful.sample.list<-successful.sample.list%>%mutate(Patient_ID=gsub(" ","", Patient_ID))%>%
  mutate(Visit_name=gsub(" ","",Visit_name))%>%
  mutate(Visit_name=gsub("BASELINE", "C1D1", Visit_name,ignore.case = TRUE))%>%
  mutate(Visit_name=gsub("UNSCH","Unscheduled",Visit_name))%>%
  mutate(Visit_name=gsub("UNS","Unscheduled",Visit_name))%>%
  mutate(Visit_name=gsub("Day1","D1", Visit_name))%>%
  mutate(Visit_name=gsub("FSV","EOS", Visit_name))

# a list of all the successful visits
Visit.check<-paste(successful.sample.list$Patient_ID, successful.sample.list$Visit_name)

Visit.check<-Visit.check[!duplicated(Visit.check)]

# fail.sample.list<-read_excel("C:\\Users\\ShaYang\\OneDrive - Stat4ward\\Documents\\Deciphera\\Phase I\\GIST\\Copy of 20210209_Deciphera_SOW1_1064Samples_G360Report_Successful_Failed.xlsx", sheet="Failed")
# 
# fail.sample.list<-fail.sample.list%>%mutate(Patient_ID=gsub(" ","", Patient_ID))%>%
#   mutate(Visit_name=gsub(" ","",Visit_name))%>%
#   mutate(Visit_name=gsub("BASELINE", "C1D1", Visit_name,ignore.case = TRUE))%>%
#   mutate(Visit_name=gsub("UNSCH","Unscheduled",Visit_name))%>%
#   mutate(Visit_name=gsub("UNS","Unscheduled",Visit_name))%>%
#   mutate(Visit_name=gsub("Day1","D1", Visit_name))%>%
#   mutate(Visit_name=gsub("FSV","EOS", Visit_name))


```

ctDNA data from Guardant Health (20210319_Deciphera_SOW1_1064Samples_G360Report_CumulativeLATEST.csv)
```{r}

ctDNA<-ctDNA%>%mutate(Patient_ID=gsub(" ","", Patient_ID))

ctDNA<-ctDNA%>%mutate(Patient_ID=ifelse(Patient_ID=="081-02","08-102",Patient_ID))

no.patient<-n_distinct(ctDNA$Patient_ID)

ctDNA<-ctDNA%>%filter(Alt.NoAlt=="1") 

# "Alt-NoAlt" = 1 means that at least 1 somatic mutation was detected in that sample

# ctDNA1<-ctDNA%>%
#   filter(Sample_status=="SUCCESS")
# 
# no.row.fail<-nrow(ctDNA)-nrow(ctDNA1)

ctDNA.germline<-nrow(ctDNA%>%filter(Somatic_status=="germline"))

# ctDNA1<-ctDNA1%>%
#   filter(Somatic_status=="somatic")  will update this once the somatic status is corrected

 ctDNA1<-ctDNA%>%
   filter(Somatic_status!="germline")  

   
#no.patient.somatic<-n_distinct(ctDNA1$Patient_ID)

#unique(ctDNA1$Variant_type)

#ctDNA1%>%filter(Variant_type=="Fusion")

#ctDNA.fusion<-ctDNA1%>%filter(Variant_type=="Fusion")

#ctDNA1<-ctDNA1%>%filter(Variant_type!="Fusion")%>%droplevels()

no.patient.sucess<-n_distinct(ctDNA1$Patient_ID)

ctDNA1<-ctDNA1%>%mutate(Visit_name=gsub(" ","",Visit_name))%>%
  mutate(Visit_name=gsub("BASELINE", "C1D1", Visit_name,ignore.case = TRUE))%>%
  mutate(Visit_name=gsub("UNSCH","Unscheduled",Visit_name))%>%
  mutate(Visit_name=gsub("UNS","Unscheduled",Visit_name))%>%
  mutate(Visit_name=gsub("Day1","D1", Visit_name))%>%
  mutate(Visit_name=gsub("FSV","EOS", Visit_name))

ctDNA1<-ctDNA1%>%mutate(Visit_name=ifelse(Visit_name%in%c("3","4","8")&str_detect(Customer_SampleID, "BASE"),"C1D1",Visit_name))

ctDNA1<-ctDNA1%>%mutate(Received_date=as.Date(Received_date, "%m/%d/%Y"),
                   Bloodcoll_date=as.Date(Bloodcoll_date, "%m/%d/%Y"),
                   Reported_date=as.Date(Reported_date,"%m/%d/%Y"))

##???
ctDNA1<-ctDNA1%>%mutate(Cancer.type=case_when(
  Cancertype=="GIST (Gastrointestinal Stromal Tumor)"~ "GIST",
  Cancertype=="GIST (Gastrointestinal stromal tumor)"~ "GIST",
  Cancertype=="Glioblastoma"~ "GBM",
  TRUE ~ Cancertype))

ctDNA1<-ctDNA1%>%mutate(Cancer.type = na_if(Cancer.type, ""))
```

There are `r nrow(ctDNA)` rows in the original ctDNA data set for Cancertype GIST (Gastrointestinal stromal tumor). `r no.row.fail` rows are excluded since Sample Status is NEGATIVE, `r n_distinct(ctDNA$Patient_ID)-n_distinct(ctDNA1$Patient_ID)` patients are eliminated because of this step. There are  `r no.patient.sucess` patients in the data with Sample Status being Success. 

```{r}

# visit.summary<-as.data.frame.matrix(table(ctDNA1$Patient_ID, ctDNA1$Visit_name))

myVars = c("Variant_type","Indel_type","Gene","Exon","Percentage","Received_date")
catVars= c("Variant_type","Indel_type","Gene","Exon","Received_date")
tab.summary <- CreateTableOne(vars = myVars, data = ctDNA1, factorVars = catVars)

```

```{r}
ID.n.adsl<-n_distinct(adsl$SUBJID)

myVars = c("COUNTRY","AGE","SEX","PHASE","CANTYP","GISTLINE","GENE1","GISTMUT")
catVars = c("COUNTRY","SEX","PHASE","CANTYP","GISTLINE","GENE1","GISTMUT")
adsl.summary<-CreateTableOne(vars = myVars, data = adsl, factorVars = catVars)


```

```{r}
ID.n.adtte<-n_distinct(adtte$SUBJID)

myVars = c("COUNTRY","AGE","SEX","RACE","ETHNIC","DOSEGR","DOSEGRN","GISTLINE","ARM","ARMCD",
           "PARAMCD","AVAL","CNSR","EVNTDESC","PHASE","GISTMUT")
catVars = c("COUNTRY","SEX","RACE","ETHNIC","DOSEGR","DOSEGRN","GISTLINE","ARM","ARMCD",
           "PARAMCD","CNSR","EVNTDESC","PHASE","GISTMUT")
adtte.summary<-CreateTableOne(vars = myVars, data = adtte, factorVars = catVars)

print(adtte.summary, showAllLevels = TRUE, formatOptions = list(big.mark = ","))

adtte.PFS.summary<-CreateTableOne(vars = myVars, data = adtte%>%filter(GISTLINE!="2nd Line"&GISTLINE!="3rd Line")%>%filter(PARAMCD=="PFS")%>%filter(DOSEGR=="150 mg QD"|DOSEGR=="150 mg BID"), factorVars = catVars)

# print(adtte.PFS.summary, showAllLevels = TRUE, formatOptions = list(big.mark = ","))

```

```{r}

#Imputations

ctDNA1<-ctDNA1%>%group_by(Patient_ID)%>%fill(Cancer.type, .direction = "updown")%>%ungroup()

ctDNA.mutation<-ctDNA1%>%
  filter(Variant_type!="CNV")%>%
  #mutate(Visit_name=droplevels(as.factor(Visit_name)), Gene=droplevels(as.factor(Gene)), Exon=droplevels(as.factor(Exon)))%>%
  expand(Visit_name,nesting(Patient_ID,Variant_type,Gene, Exon, Mut_aa, Mut_nt))%>%arrange(Patient_ID,Visit_name,Variant_type,Gene,Exon)


impute<-ctDNA.mutation%>%left_join(ctDNA1, by = c("Patient_ID"="Patient_ID",
                                                        "Visit_name"="Visit_name",
                                                        "Variant_type"="Variant_type",
                                                        "Gene"="Gene",
                                                        "Exon"="Exon",
                                                        "Mut_aa"="Mut_aa",
                                                        "Mut_nt"="Mut_nt"))%>%
  group_by(Patient_ID,Variant_type,Gene,Exon,Mut_aa,Mut_nt)%>%
  fill(`ï..Study_ID`:Transcript, .direction = "updown")%>%
  fill(Cancer.type, .direction = "updown")%>%
  ungroup()%>%
  mutate(Percentage=case_when(
    !is.na(Percentage)~Percentage,
    is.na(Percentage)&paste(Patient_ID,Visit_name)%in%Visit.check ~ 0.02,
    is.na(Percentage)&!paste(Patient_ID,Visit_name)%in%Visit.check ~ 999))%>%
  filter(Percentage!=999)

# %>%mutate(Visit_name=
#         factor(Visit_name,levels =       c("C1D1","C3D1","C5D1","C5D15","C7D1","C9D1","C11D1","C12D1","COC1D1","COC3D1","COC5D1","COC7D1","COC9D1","EOT")) )

cnv<-ctDNA1%>%
  filter(Variant_type=="CNV")%>%select("Visit_name","Patient_ID",         
  "Variant_type",        "Gene",               
  "Exon",                "Mut_aa" ,            
  "Mut_nt" ,             "ï..Study_ID",        
 "Customer_SampleID",   "GHRequestID",        
 "Alt.NoAlt" :"Cancer.type"   )

impute.all<-bind_rows(impute, cnv)%>%select(-"Cancertype")
```

```{r}
 t<-adtte%>%select(SUBJID, GISTLINE, ARM, PARAM,PARAMCD, AVAL,AVALC, CNSR, "EVNTDESC","PHASE","GISTMUT")%>%filter(PARAMCD=="PFS")

impute.clinical<-impute.all%>%mutate(ID=gsub("-","", Patient_ID))%>%left_join(t, by=c("ID"="SUBJID"))


no.clinical<-impute.clinical%>%filter(is.na(GISTMUT))%>%select(Patient_ID, Gene,Exon,Cancer.type)%>%distinct()

write.csv(impute.clinical, "imputation_with_clinical.csv")

write.csv(no.clinical, "Pts_wo_clinical.csv")

```
